frontend/src/components/utils/sidebar/CheckboxField.tsx

```diff
@@ -1,13 +1,17 @@
 import React from "react";
 import { FormCheck } from "react-bootstrap";
 
-export const CheckboxField = ({ label, state, setState }) => (
+export const CheckboxField = ({ label, state, setState, id }) => (
     <label className="custom-control custom-checkbox">
         <FormCheck.Input
             className="custom-control-input"
             type="checkbox"
-            checked={state}
-            onChange={() => setState(!state)}
+            checked={state[id]}
+            onChange={() => {
+                let newState = { ...state };
+                newState[id] = !state[id];
+                setState(newState);
+            }}
```

Comment:
```suggestion
            onChange={() => setState({
                ...state,
                [id]: !state[id]
            })};
```
or something like this.
https://stackoverflow.com/a/2274327

Reply:
Niceeee